File: week2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int num_letters(string text)
{
    int letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters += 1;
        }
    }
    return letters;
}
int num_words(string text)
{
    int words = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ' && text[i - 1] != ' ')
        {
            words += 1;
        }
    }
    return words;
}
int num_sentences(string text)
{
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences += 1;
        }
    }
    return sentences;
}
float letters_per_words(int letters, int words)
{
    return ((float) letters / words * 100);
}
float sentences_per_words(int sentences, int words)
{
    return ((float) sentences / words * 100);
}
int main(void)
{
    string text = get_string("Text: ");
    int words = num_words(text);
    int letters = num_letters(text);
    int sentences = num_sentences(text);
    float sentences_per = sentences_per_words(sentences, words);
    float letters_per = letters_per_words(letters, words);
    int index = round(0.0588 * letters_per - 0.296 * sentences_per - 15.8);
    if (index > 1 && index < 16)
    {
        printf("Grade %d\n", index);
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }
}

File: week6/readability.py
def num_letters(text):
    letters = 0
    ltext = list(text)
    for i in range(len(text)):
        if (ltext[i].isalpha()):
            letters += 1
    return letters


def num_sentences(text):
    sentences = 0
    ltext = list(text)
    for i in range(len(text)):
        if (ltext[i] == '.' or ltext[i] == '?' or ltext[i] == '!'):
            sentences += 1
    return sentences


def num_words(text):
    return len(text.split())


def letters_per_words(letters, words):
    return (letters / words) * 100


def sentences_per_words(sentences, words):
    return (sentences / words) * 100


text = str(input("Text: "))
words = num_words(text)
letters = num_letters(text)
sentences = num_sentences(text)
sentences_per = sentences_per_words(sentences, words)
letters_per = letters_per_words(letters, words)
index = round(0.0588 * letters_per - 0.296 * sentences_per - 15.8)
if (index > 1 and index < 16):
    print("Grade", index)
elif (index < 1):
    print("Before Grade 1\n")
else:
    print("Grade 16+\n")
